src/main/java/com/projectg/geyserupdater/common/util/GeyserApi.java

```diff
@@ -0,0 +1,32 @@
+package com.projectg.geyserupdater.common.util;
+
+import com.google.gson.Gson;
+import com.projectg.geyserupdater.common.pojo.Root;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class GeyserApi {
+    public Root endPoints() {
+        try {
+            URL url = new URL(Constants.GEYSER_BASE_URL + Constants.GEYSER_LATEST_MASTER_ENDPOINT);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+
+            int responseCode = connection.getResponseCode();
+            if (responseCode == HttpURLConnection.HTTP_OK) {
+                Gson gson = new Gson();
```

Comment:
This should be saved to a static field

Reply:
We never access the Gson anywhere else to thought it would be fine to just grab an instance there like that

Reply:
stuff like Gson shouldn't be reconstructed every time its needed